src/DynamoCore/Models/RecordableCommands.cs

```diff
@@ -504,6 +504,20 @@ internal static OpenFileCommand DeserializeCore(XmlElement element)
             protected override void ExecuteCore(DynamoModel dynamoModel)
             {
                 dynamoModel.OpenFileImpl(this);
+
+                // Log file open action and the number of nodes in the opened workspace
+                Dynamo.Logging.Analytics.CreateFileOperationEvent(
+                    XmlFilePath, Logging.Actions.Open, dynamoModel.CurrentWorkspace.Nodes.Count());
+
+                // Log unresolved node names and node count in the opened workspace
+                var unresolvedNodes = dynamoModel.CurrentWorkspace.Nodes.OfType<DummyNode>();
+                if (unresolvedNodes != null && unresolvedNodes.Any())
+                {
+                    Dynamo.Logging.Analytics.TrackEvent(Logging.Actions.Unresolved, Logging.Categories.NodeOperations,
```

Comment:
This needs to go to instrumentation not analytics as the assembly could contain names.